assets/app/features/upgrade/controllers/upgrade-database-configuration.controller.js

```diff
@@ -11,16 +11,92 @@
     angular.module('crowbarApp')
         .controller('UpgradeDatabaseConfigurationController', UpgradeDatabaseConfigurationController);
 
-    UpgradeDatabaseConfigurationController.$inject = [];
+    UpgradeDatabaseConfigurationController.$inject = ['upgradeFactory'];
     // @ngInject
-    function UpgradeDatabaseConfigurationController() {
+    function UpgradeDatabaseConfigurationController(upgradeFactory) {
         var vm = this;
         vm.databaseForm = {
             username: '',
             password: '',
             server: '',
-            port: 5432,
-            tablePrefix: ''
+            port: 5432
+        };
+
+        vm.running = false;
+        vm.spinnerVisible = false;
+        vm.completed = false;
+        vm.errors = [];
+
+        vm.createServer = createServer;
+        vm.connectServer = connectServer;
+
+        // functions
+
+        function createServer() {
+            var emptyFields = _.some(
```

Comment:
Leave it as it is if that makes your life and development easier. :) Just know that this will be reviewed/adjusted when we work on the other cards.